src/CoreFoundation/DispatchBlock.cs

```diff
@@ -15,8 +15,107 @@
+				block_handler.CleanupBlock ();
+				return ret;
+			}
+		}
+
+		[DllImport (Constants.libcLibrary)]
+		extern static IntPtr dispatch_block_create (DispatchBlockFlags flags, IntPtr block);
+		
+		public DispatchBlock Create (Action action, DispatchBlockFlags flags)
+		{
+			unsafe {
+			        BlockLiteral block_handler = new BlockLiteral ();
+			        BlockLiteral *block_ptr_handler = &block_handler;
+			        block_handler.SetupBlockUnsafe (BlockStaticDispatchClass.static_dispatch_block, action);
+				var ret =  _Block_copy (block_ptr_handler);
+				block_handler.CleanupBlock ();
+				return new DispatchBlock (dispatch_block_create (flags, ret));
+			}
+		}
+
+		[DllImport (Constants.libcLibrary)]
+		extern static IntPtr dispatch_block_create_with_qos_class (DispatchBlockFlags flags, DispatchQualityOfService qosClass, int relative_priority, IntPtr dispatchBlock);
+
+		public DispatchBlock CreateWithQos (DispatchBlockFlags flags, DispatchQualityOfService qosClass, int relative_priority, Action action)
+		{
+			unsafe {
+			        BlockLiteral block_handler = new BlockLiteral ();
+			        BlockLiteral *block_ptr_handler = &block_handler;
+			        block_handler.SetupBlockUnsafe (BlockStaticDispatchClass.static_dispatch_block, action);
+				var ret = _Block_copy (block_ptr_handler);
+				block_handler.CleanupBlock ();
+				return new DispatchBlock (dispatch_block_create_with_qos_class (flags, qosClass, relative_priority, ret));
+			}
+		}
+
+		public DispatchBlock CreateWithQos (DispatchBlockFlags flags, DispatchQualityOfService qosClass, int relative_priority, DispatchBlock dispatchBlock)
+		{
+			if (dispatchBlock == null)
+				throw new ArgumentNullException (nameof (dispatchBlock));
+			return new DispatchBlock (dispatch_block_create_with_qos_class (flags, qosClass, relative_priority, dispatchBlock.blockHandle));
+		}
+
+		[DllImport (Constants.libcLibrary)]
+		extern static void dispatch_block_cancel (IntPtr block);
+		
+		public void Cancel ()
+		{
+			dispatch_block_cancel (blockHandle);
+		}
+
+		[DllImport (Constants.libcLibrary)]
+		extern static void dispatch_block_notify (IntPtr block, IntPtr queue, IntPtr notification);
+		
+		public void Notify (DispatchQueue queue, DispatchBlock notification)
+		{
+			if (queue == null)
+				throw new ArgumentNullException (nameof (queue));
+			if (notification == null)
+				throw new ArgumentNullException (nameof (notification));
+			dispatch_block_notify (blockHandle, queue.Handle, notification.blockHandle);
+		}
+		
+		[DllImport (Constants.libcLibrary)]
+		extern static long dispatch_block_testcancel(IntPtr block);
+
+		public long TestCancel ()
+		{
+			return dispatch_block_testcancel (blockHandle);
+		}
+
+		[DllImport (Constants.libcLibrary)]
+		extern static long dispatch_block_wait (IntPtr block, DispatchTime time);
+
+		public void Wait (DispatchTime time)
+		{
+			dispatch_block_wait (blockHandle, time);
```

Comment:
this hides the return value (i.e. it should also return `nint`)

Reply:
Fixed.